crates/rolldown_common/src/inner_bundler_options/types/filename_template.rs

```diff
@@ -33,7 +34,7 @@ impl FilenameTemplate {
       tmp = tmp.replace("[name]", name);
     }
     if let Some(hash) = options.hash {
-      tmp = tmp.replace("[hash]", hash);
+      tmp = tmp.replace(options.hash_pattern.unwrap_or("[hash]"), hash);
```

Comment:
I recommend completing the task in the file_emitter.rs file. For reference, the relevant code can be found [here](https://github.com/rolldown/rolldown/blob/main/crates/rolldown_common/src/file_emitter.rs#L86).